device_doctor/test/src/ios_device_test.dart

```diff
@@ -42,31 +44,119 @@ void main() {
+      StringBuffer sb = new StringBuffer();
+      sb.writeln('1) abcdefg "Apple Development: Flutter Devicelab (hijklmn)"');
+      sb.writeln('1 valid identities found');
+      output = <List<int>>[utf8.encode(sb.toString())];
+      process = FakeProcess(0, out: output);
+      HealthCheckResult healthCheckResult = await deviceDiscovery.certCheck(processManager: processManager);
+      expect(healthCheckResult.succeeded, true);
+    });
+
+    test('Cert check - failure', () async {
+      when(processManager.start(any, workingDirectory: anyNamed('workingDirectory')))
+          .thenAnswer((_) => Future.value(process));
+      StringBuffer sb = new StringBuffer();
+      sb.writeln('abcdefg');
+      sb.writeln('hijklmn');
+      output = <List<int>>[utf8.encode(sb.toString())];
+      process = FakeProcess(0, out: output);
+      HealthCheckResult healthCheckResult = await deviceDiscovery.certCheck(processManager: processManager);
+      expect(healthCheckResult.succeeded, false);
+      expect(healthCheckResult.name, kCertCheckKey);
+      expect(healthCheckResult.details, sb.toString().trim());
+    });
+
+    test('Cert check - exeption', () async {
+      when(processManager.start(any, workingDirectory: anyNamed('workingDirectory')))
+          .thenAnswer((_) => Future.value(process));
+      process = FakeProcess(1);
+      HealthCheckResult healthCheckResult = await deviceDiscovery.certCheck(processManager: processManager);
+      expect(healthCheckResult.succeeded, false);
+      expect(healthCheckResult.name, kCertCheckKey);
+      expect(healthCheckResult.details, 'Executable security failed with exit code 1.');
+    });
+
+    test('Device pair check - success', () async {
+      when(processManager.start(any, workingDirectory: anyNamed('workingDirectory')))
+          .thenAnswer((_) => Future.value(process));
+      StringBuffer sb = new StringBuffer();
+      sb.writeln('SUCCESS: Validated pairing with device abcdefg-hijklmn');
+      output = <List<int>>[utf8.encode(sb.toString())];
+      process = FakeProcess(0, out: output);
+      HealthCheckResult healthCheckResult = await deviceDiscovery.devicePairCheck(processManager: processManager);
+      expect(healthCheckResult.succeeded, true);
+    });
+
+    test('Device pair check - failure', () async {
+      when(processManager.start(any, workingDirectory: anyNamed('workingDirectory')))
+          .thenAnswer((_) => Future.value(process));
+      StringBuffer sb = new StringBuffer();
+      sb.writeln('abcdefg');
+      output = <List<int>>[utf8.encode(sb.toString())];
+      process = FakeProcess(0, out: output);
+      HealthCheckResult healthCheckResult = await deviceDiscovery.devicePairCheck(processManager: processManager);
+      expect(healthCheckResult.succeeded, false);
+      expect(healthCheckResult.name, kDevicePairCheckKey);
+      expect(healthCheckResult.details, sb.toString().trim());
+    });
+
+    test('Device pair check - exeption', () async {
```

Comment:
Can we add a test with more than one valid certificate?

Reply:
Added.